web/src/test/java/com/replica/RouterServerTest.java

```diff
@@ -640,6 +646,45 @@ public void testPathDetailsReturnedOnlyWhenRequested() {
         }
     }
 
+    @Test
+    public void testDuplicateRoutesFiltering() {
+        // When duplicate routes are allowed, we get 1 route for each profile in CAR_PROFILES
+        final RouterOuterClass.StreetRouteReply responseWithDuplicates = routerStub.routeStreetMode(
+                createStreetRequest("car", false, REQUEST_ORIGIN_1, REQUEST_DESTINATION_1, false, true));
+        assertEquals(CAR_PROFILES.size(), responseWithDuplicates.getPathsList().size());
+
+        // When duplicate route filtering is on, routes from 2 profiles are removed due to redundancy with other routes
+        Set<String> expectedProfilesAfterDuplicatesFiltered = Sets.newHashSet(CAR_PROFILES);
+        expectedProfilesAfterDuplicatesFiltered.removeAll(Set.of(CUSTOM_THURTON_DRIVE_CAR_PROFILE_NAME, CLOSED_BASELINE_ROAD_CAR_PROFILE_NAME));
+        final RouterOuterClass.StreetRouteReply responseWithoutDuplicates = routerStub.routeStreetMode(
+                createStreetRequest("car", false, REQUEST_ORIGIN_1, REQUEST_DESTINATION_1, false, false));
+        assertEquals(expectedProfilesAfterDuplicatesFiltered.size(), responseWithoutDuplicates.getPathsList().size());
```

Comment:
i see we check the size instead of the exact profiles returned post-deduping...do we have guarantees about which profiles take priority? if so it seems worth checking the profiles themselves. if not, seems like we should just check the size and not bother creating a `Set<String> expectedProfilesAfterDuplicatesFiltered` with full profile names that are later ignored

Reply:
Good point, I did kind of over-specify what we're checking here if all we're comparing is the size of each profile set.

We do have loose guarantees about which profiles take priority, but as far as I can tell, it's dependent on the order that the profiles are defined in `test_gh_config.yaml`; if those changed, then the order we see routes in the result set changes, and we potentially filter out a different set of duplicate routes (coming from different profiles).

I think the danger from this potential inconsistency is fairly small, so I updated here to check equality of the actual profile sets, instead of just the size; if something changes in the future with the profiles we have in `test_gh_config`, this test should fail loudly + force us to change the explicit profile set we're comparing against here, which seems reasonable to me

I'm happy to switch to the other method (only checking size, without creating `expectedProfilesAfterDuplicatesFiltered`) if you think that would be preferable, though! It's a bit less specific of a test, but might be more flexible to changes in profile ordering in our test config yaml

Reply:
i like the more thorough check you switched to! but to make it more obvious how to fix the test if the order does change, i think it'd be helpful to add a code comment (either in the test or in the deduping impl) noting that the "winning" profile depends on the order in the GH config yaml

Reply:
Good call! Added a comment in the test to clarify that what gets filtered out depends on profile ordering in configs